### PR TITLE
Use yarn for builds

### PR DESCRIPTION
**Pull request by akosyakov**

### CODE REVIEW

**Comment by akosyakov** [comment]
I think we cannot use yarn, it has the same issues like npm with hoisting. It only can work if we redeclare all hoisted dependencies in the root, but then we can use npm5.

**Comment by akosyakov** [comment]
To rewrite the lock file from the scratch call: `yarn install --force`

**Comment by akosyakov** [comment]
@hexa00 good to merge?

File: .travis.yml [hunk 1]
```diff
@@ -19,9 +26,11 @@ addons:
 before_script:
   - export DISPLAY=:99.0; sh -e /etc/init.d/xvfb start
 before_install:
-  - "npm install -g npm@^4"
+  - curl -o- -L https://yarnpkg.com/install.sh | bash
+  - export PATH=$HOME/.yarn/bin:$PATH
   - "google-chrome-stable --headless --disable-gpu --remote-debugging-port=9222 http://localhost &"
-bundler_args: --retry 0
+install: yarn --frozen-lockfile
```
Comment (hexa00) [inline .travis.yml]: --frozen-lockfile', "don't generate a lockfile and fail if an update is needed 
I don't think we want to fail the build if we need to update ?
Shouldn't we use the lockfile default

File: generator-theia/package.json [hunk 2]
```diff
@@ -11,14 +11,13 @@
     "yeoman-generator"
   ],
   "scripts": {
-    "prepublishOnly": "npm run clean && npm run compile",
+    "prepare": "npm run clean && npm run compile",
```
Comment (hexa00) [inline generator-theia/package.json]: We should change this to yarn also?

File: scripts/clean-all.sh [hunk 3]
```diff
@@ -1,4 +0,0 @@
-#!/bin/bash
-npm install lerna \
-lerna clean --yes \
-rm -rf node_modules
```
Comment (hexa00) [inline scripts/clean-all.sh]: Why is this gone?

File: scripts/rebuild.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
+/*
```
Comment (hexa00) [inline scripts/rebuild.js]: nice addition :)

File: scripts/update-shrinkwrap.sh [hunk 5]
```diff
@@ -1,5 +0,0 @@
-#!/bin/bash
-./scripts/clean-all.sh
-rm npm-shrinkwrap.json
-npm install
-npm shrinkwrap
```
Comment (hexa00) [inline scripts/update-shrinkwrap.sh]: Would be nice to have the yarn equivalent ?

File: scripts/clean-all.sh (hunk 3 above)
Comment (akosyakov) [inline scripts/clean-all.sh]: When do you use it?

File: scripts/update-shrinkwrap.sh (hunk 5 above)
Comment (akosyakov) [inline scripts/update-shrinkwrap.sh]: `yarn install --force`

File: scripts/clean-all.sh (hunk 3 above)
Comment (hexa00) [inline scripts/clean-all.sh]: Guess it was mainly for the shrinkwrap thing seems like we don't need it anymore with yarn install --force indeed

**Comment by hexa00** [comment]
btw in general, while I understand how you usually work by adding one small commit after the other I think it would be nicer if you could amend commits instead before sending a non WIP PR ?

This way when looking at the commit history logical changes are consistent and would not have parts of them reverted back one or 2 commits later like 'removed --frozen-lockfile' does ?

File: scripts/clean-all.sh (hunk 3 above)
Comment (kittaakos) [inline scripts/clean-all.sh]: I added it long ago, and someone has adapted it to the current structure. Obsolete.

**Comment by akosyakov** [comment]
@hexa00 I was lazy to push it this time and just changed on the github :) if the change can be removed it by amending the previous commit I usually do it, with big PRs it can be hard to amend previous done several days ago

**Comment by akosyakov** [comment]
@hexa00 squashed to one commit

**Comment by hexa00** [comment]
OK thx :)

**Pull request merged**